diff --git a/target/linux/ipq806x/base-files/etc/board.d/02_network b/target/linux/ipq806x/base-files/etc/board.d/02_network
--- a/target/linux/ipq806x/base-files/etc/board.d/02_network	2021-08-13 03:43:31.532528359 +0000
+++ b/target/linux/ipq806x/base-files/etc/board.d/02_network	2021-08-13 04:04:11.388093624 +0000
@@ -11,8 +11,7 @@
 board=$(board_name)
 
 case "$board" in
-askey,rt4230w-rev6 |\
-askey,rt4230w-rev9.3 |\
+askey,rt4230w |\
 asrock,g10 |\
 nec,wg2600hp)
 	ucidef_add_switch "switch0" \
diff --git a/target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata b/target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata
--- a/target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata	2021-08-13 03:43:31.632528085 +0000
+++ b/target/linux/ipq806x/base-files/etc/hotplug.d/firmware/11-ath10k-caldata	2021-08-13 04:05:44.880266892 +0000
@@ -25,10 +25,6 @@
 	;;
 "ath10k/pre-cal-pci-0000:01:00.0.bin")
 	case $board in
-	askey,rt4230w-rev6 |\
-	askey,rt4230w-rev9.3)
-		caldata_extract "0:ART" 0x1000 0x2f20
-		;;
 	asrock,g10)
 		if [ -b "$(find_mtd_part 0:art)" ]; then
 			caldata_extract "0:art" 0x1000 0x2f20
@@ -59,6 +55,7 @@
 		caldata_extract "ART" 0x1000 0x2f20
 		ath10k_patch_mac $(mtd_get_mac_binary PRODUCTDATA 0x12)
 		;;
+	askey,rt4230w |\
 	netgear,d7800 |\
 	netgear,r7500v2 |\
 	netgear,r7800 |\
@@ -89,10 +86,6 @@
 	;;
 "ath10k/pre-cal-pci-0001:01:00.0.bin")
 	case $board in
-	askey,rt4230w-rev6 |\
-	askey,rt4230w-rev9.3)
-		caldata_extract "0:ART" 0x5000 0x2f20
-		;;
 	asrock,g10)
 		if [ -b "$(find_mtd_part 0:art)" ]; then
 			caldata_extract "0:art" 0x5000 0x2f20
@@ -115,6 +108,7 @@
 		caldata_extract "ART" 0x5000 0x2f20
 		ath10k_patch_mac $(mtd_get_mac_binary PRODUCTDATA 0xc)
 		;;
+	askey,rt4230w |\
 	netgear,d7800 |\
 	netgear,r7500v2 |\
 	netgear,r7800 |\
diff --git a/target/linux/ipq806x/base-files/lib/upgrade/platform.sh b/target/linux/ipq806x/base-files/lib/upgrade/platform.sh
--- a/target/linux/ipq806x/base-files/lib/upgrade/platform.sh	2021-08-13 03:43:31.632528085 +0000
+++ b/target/linux/ipq806x/base-files/lib/upgrade/platform.sh	2021-08-13 04:04:30.176131854 +0000
@@ -10,8 +10,7 @@
 
 platform_do_upgrade() {
 	case "$(board_name)" in
-	askey,rt4230w-rev6 |\
-	askey,rt4230w-rev9.3 |\
+	askey,rt4230w |\
 	compex,wpq864 |\
 	netgear,d7800 |\
 	netgear,r7500 |\
diff --git a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dts b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dts
--- a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dts	1970-01-01 00:00:00.000000000 +0000
+++ b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dts	2021-08-13 03:52:30.975353781 +0000
@@ -0,0 +1,395 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "qcom-ipq8065.dtsi"
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "Askey RT4230W";
+	compatible = "askey,rt4230w", "qcom,ipq8065", "qcom,ipq8064";
+
+	memory@0 {
+		reg = <0x42000000 0x3e000000>;
+		device_type = "memory";
+	};
+
+	reserved-memory {
+		#address-cells = <1>;
+		#size-cells = <1>;
+		ranges;
+		rsvd@41200000 {
+			reg = <0x41200000 0x300000>;
+			no-map;
+		};
+	};
+
+	aliases {
+		serial0 = &gsbi4_serial;
+		mdio-gpio0 = &mdio0;
+		led-boot = &ledctrl3;
+		led-running = &ledctrl2;
+	};
+
+	chosen {
+		bootargs = "rootfstype=squashfs noinitrd";
+		stdout-path = "serial0:115200n8";
+	};
+
+	keys {
+		compatible = "gpio-keys";
+		pinctrl-0 = <&button_pins>;
+		pinctrl-names = "default";
+
+		reset {
+			label = "reset";
+			gpios = <&qcom_pinmux 54 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+			debounce-interval = <60>;
+			wakeup-source;
+		};
+
+		wps {
+			label = "wps";
+			gpios = <&qcom_pinmux 68 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_WPS_BUTTON>;
+			debounce-interval = <60>;
+			wakeup-source;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-0 = <&led_pins>;
+		pinctrl-names = "default";
+
+		ledctrl1: ledctrl1 {
+			label = "rt4230w:ledctrl:1";
+			gpios = <&qcom_pinmux 22 GPIO_ACTIVE_HIGH>;
+		};
+		
+		ledctrl2: ledctrl2 {
+			label = "rt4230w:ledctrl:2";
+			gpios = <&qcom_pinmux 23 GPIO_ACTIVE_HIGH>;
+		};
+		
+		ledctrl3: ledctrl3 {
+			label = "rt4230w:ledctrl:3";
+			gpios = <&qcom_pinmux 24 GPIO_ACTIVE_HIGH>;
+		};
+	};
+};
+
+&qcom_pinmux {
+	button_pins: button_pins {
+		mux {
+			pins = "gpio54", "gpio68";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-up;
+		};
+	};
+
+	i2c4_pins: i2c4_pinmux {
+		mux {
+			pins = "gpio12", "gpio13";
+			function = "gsbi4";
+			drive-strength = <12>;
+			bias-disable;
+		};
+	};
+
+	led_pins: led_pins {
+		mux {
+			pins = "gpio22", "gpio23", "gpio24";
+			function = "gpio";
+			drive-strength = <2>;
+			bias-pull-down;
+		};
+	};
+
+	nand_pins: nand_pins {
+		disable {
+			pins = "gpio34", "gpio35", "gpio36",
+			       "gpio37", "gpio38";
+			function = "nand";
+			drive-strength = <10>;
+			bias-disable;
+		};
+
+		pullups {
+			pins = "gpio39";
+			function = "nand";
+			drive-strength = <10>;
+			bias-pull-up;
+		};
+
+		hold {
+			pins = "gpio40", "gpio41", "gpio42",
+			       "gpio43", "gpio44", "gpio45",
+			       "gpio46", "gpio47";
+			function = "nand";
+			drive-strength = <10>;
+			bias-bus-hold;
+		};
+	};
+
+	mdio0_pins: mdio0_pins {
+		mux {
+			pins = "gpio0", "gpio1";
+			function = "gpio";
+			drive-strength = <8>;
+			bias-disable;
+		};
+	};
+
+	rgmii2_pins: rgmii2_pins {
+		mux {
+			pins = "gpio27", "gpio28", "gpio29", "gpio30", "gpio31",
+					"gpio51", "gpio52", "gpio59", "gpio60", "gpio61", "gpio62" ;
+			function = "rgmii2";
+			drive-strength = <8>;
+			bias-disable;
+		};
+
+		tx {
+			pins = "gpio27", "gpio28", "gpio29", "gpio30", "gpio31", "gpio32" ;
+			input-disable;
+		};
+	};
+};
+
+&gsbi4 {
+	qcom,mode = <GSBI_PROT_I2C_UART>;
+	status = "okay";
+
+	serial@16340000 {
+		status = "okay";
+	};
+
+	/*
+	* The i2c device on gsbi4 should not be enabled.
+	* On ipq806x designs gsbi4 i2c is meant for exclusive
+	* RPM usage. Turning this on in kernel manifests as
+	* i2c failure for the RPM.
+	*/
+};
+
+&nand_controller {
+	status = "okay";
+
+	pinctrl-0 = <&nand_pins>;
+	pinctrl-names = "default";
+
+	nand@0 {
+		reg = <0>;
+		compatible = "qcom,nandcs";
+
+		nand-ecc-strength = <4>;
+		nand-bus-width = <8>;
+		nand-ecc-step-size = <512>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			qcadata@0 {
+				label = "qcadata";
+				reg = <0x0000000 0x0c80000>;
+			};
+
+			APPSBL@c80000 {
+				label = "APPSBL";
+				reg = <0x0c80000 0x0500000>;
+			};
+
+			APPSBLENV@1180000 {
+				label = "APPSBLENV";
+				reg = <0x1180000 0x0080000>;
+			};
+
+			art: art@1200000 {
+				label = "art";
+				reg = <0x1200000 0x0140000>;
+			};
+
+			artbak: art@1340000 {
+				label = "artbak";
+				reg = <0x1340000 0x0140000>;
+			};
+
+			kernel@1480000 {
+				label = "kernel";
+				reg = <0x1480000 0x0400000>;
+			};
+
+			ubi@16a0000 {
+				label = "ubi";
+				reg = <0x1880000 0xE780000>;
+			};
+
+		};
+	};
+};
+
+&mdio0 {
+        status = "okay";
+
+	compatible = "virtual,mdio-gpio";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	gpios = <&qcom_pinmux 1 GPIO_ACTIVE_HIGH>,
+		<&qcom_pinmux 0 GPIO_ACTIVE_HIGH>;
+        pinctrl-0 = <&mdio0_pins>;
+        pinctrl-names = "default";
+
+        phy0: ethernet-phy@0 {
+		device_type = "ethernet-phy";
+		reg = <0x0>;
+                qca,ar8327-initvals = <
+                        0x00004 0x7600000   /* PAD0_MODE */
+                        0x00008 0x1000000   /* PAD5_MODE */
+                        0x0000c 0x80        /* PAD6_MODE */
+                        0x000e4 0xaa545     /* MAC_POWER_SEL */
+                        0x000e0 0xc74164de  /* SGMII_CTRL */
+                        0x0007c 0x4e        /* PORT0_STATUS */
+                        0x00094 0x4e        /* PORT6_STATUS */
+                        0x00970 0x1e864443  /* QM_PORT0_CTRL0 */
+                        0x00974 0x000001c6  /* QM_PORT0_CTRL1 */
+                        0x00978 0x19008643  /* QM_PORT1_CTRL0 */
+                        0x0097c 0x000001c6  /* QM_PORT1_CTRL1 */
+                        0x00980 0x19008643  /* QM_PORT2_CTRL0 */
+                        0x00984 0x000001c6  /* QM_PORT2_CTRL1 */
+                        0x00988 0x19008643  /* QM_PORT3_CTRL0 */
+                        0x0098c 0x000001c6  /* QM_PORT3_CTRL1 */
+                        0x00990 0x19008643  /* QM_PORT4_CTRL0 */
+                        0x00994 0x000001c6  /* QM_PORT4_CTRL1 */
+                        0x00998 0x1e864443  /* QM_PORT5_CTRL0 */
+                        0x0099c 0x000001c6  /* QM_PORT5_CTRL1 */
+                        0x009a0 0x1e864443  /* QM_PORT6_CTRL0 */
+                        0x009a4 0x000001c6  /* QM_PORT6_CTRL1 */
+                        >;
+        };
+
+        phy4: ethernet-phy@4 {
+		device_type = "ethernet-phy";
+                reg = <0x4>;
+        };
+};
+
+&gmac0 {
+	status = "okay";
+	compatible = "qcom,nss-gmac";
+	reg = <0x37000000 0x200000>;
+	interrupts = <GIC_SPI 220 IRQ_TYPE_LEVEL_HIGH>;
+	phy-mode = "rgmii";
+	qcom,id = <0>;
+	qcom,pcs-chanid = <0>;
+	qcom,phy-mdio-addr = <4>;
+	qcom,poll-required = <0>;
+	qcom,rgmii-delay = <1>;
+	qcom,phy_mii_type = <0>;
+	qcom,emulation = <0>;
+	qcom,forced-speed = <1000>;
+	qcom,forced-duplex = <1>;
+	qcom,socver = <0>;
+	qcom,irq = <255>;
+	mdiobus = <&mdio0>;
+
+	pinctrl-0 = <&rgmii2_pins>;
+	pinctrl-names = "default";
+
+	mtd-mac-address = <&art 6>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+};
+
+&gmac1 {
+	status = "okay";
+	compatible = "qcom,nss-gmac";
+	reg = <0x37200000 0x200000>;
+	interrupts = <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
+	phy-mode = "sgmii";
+	qcom,id = <1>;
+	qcom,pcs-chanid = <1>;
+	qcom,phy-mdio-addr = <0>;	/* none */
+	qcom,poll-required = <0>;	/* no polling */
+	qcom,rgmii-delay = <0>;
+	qcom,phy_mii_type = <1>;
+	qcom,emulation = <0>;
+	qcom,forced-speed = <1000>;
+	qcom,forced-duplex = <1>;
+	qcom,socver = <0>;
+	qcom,irq = <258>;
+	mdiobus = <&mdio0>;
+
+	mtd-mac-address = <&art 0>;
+
+	fixed-link {
+		speed = <1000>;
+		full-duplex;
+	};
+
+};
+
+&tcsr {
+        compatible = "qcom,tcsr";
+        reg = <0x1a400000 0x100>;
+        status = "okay";
+        qcom,usb-ctrl-select = <0x03>;
+};
+
+&adm_dma {
+	status = "okay";
+};
+
+&usb3_0 {
+		status = "okay";
+		clocks = <&gcc USB30_1_MASTER_CLK>;
+};
+
+&usb3_1 {
+		status = "okay";
+		clocks = <&gcc USB30_0_MASTER_CLK>;
+};
+
+&pcie0 {
+	status = "okay";
+
+	bridge@0,0 {
+		reg = <0x00000000 0 0 0 0>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		ranges;
+
+		wifi@1,0 {
+			compatible = "pci168c,0046";
+			reg = <0x00010000 0 0 0 0>;
+
+			mtd-mac-address = <&art 6>;
+			mtd-mac-address-increment = <(1)>;
+		};
+	};
+};
+
+&pcie1 {
+	status = "okay";
+	force_gen1 = <1>;
+
+	bridge@0,0 {
+		reg = <0x00000000 0 0 0 0>;
+		#address-cells = <3>;
+		#size-cells = <2>;
+		ranges;
+
+		wifi@1,0 {
+			compatible = "pci168c,0046";
+			reg = <0x00010000 0 0 0 0>;
+
+			mtd-mac-address = <&art 6>;
+			mtd-mac-address-increment = <(2)>;
+		};
+	};
+};
diff --git a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dtsi b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dtsi
--- a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dtsi	2021-08-13 03:43:31.636528074 +0000
+++ b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w.dtsi	1970-01-01 00:00:00.000000000 +0000
@@ -1,364 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-or-later
-
-#include "qcom-ipq8065.dtsi"
-#include <dt-bindings/input/input.h>
-
-/ {
-	memory@0 {
-		reg = <0x42000000 0x3e000000>;
-		device_type = "memory";
-	};
-
-	aliases {
-		led-boot = &ledctrl3;
-		led-failsafe = &ledctrl1;
-		led-running = &ledctrl2;
-		led-upgrade = &ledctrl3;
-	};
-
-	chosen {
-		bootargs = "rootfstype=squashfs noinitrd";
-	};
-
-	keys {
-		compatible = "gpio-keys";
-		pinctrl-0 = <&button_pins>;
-		pinctrl-names = "default";
-
-		reset {
-			label = "reset";
-			gpios = <&qcom_pinmux 54 GPIO_ACTIVE_LOW>;
-			linux,code = <KEY_RESTART>;
-		};
-
-		wps {
-			label = "wps";
-			gpios = <&qcom_pinmux 68 GPIO_ACTIVE_LOW>;
-			linux,code = <KEY_WPS_BUTTON>;
-		};
-	};
-
-	leds {
-		compatible = "gpio-leds";
-		pinctrl-0 = <&led_pins>;
-		pinctrl-names = "default";
-
-		ledctrl1: ledctrl1 {
-			label = "ledctrl1";
-			gpios = <&qcom_pinmux 22 GPIO_ACTIVE_HIGH>;
-		};
-
-		ledctrl2: ledctrl2 {
-			label = "ledctrl2";
-			gpios = <&qcom_pinmux 23 GPIO_ACTIVE_HIGH>;
-		};
-
-		ledctrl3: ledctrl3 {
-			label = "ledctrl3";
-			gpios = <&qcom_pinmux 24 GPIO_ACTIVE_HIGH>;
-		};
-	};
-};
-
-&qcom_pinmux {
-	button_pins: button_pins {
-		mux {
-			pins = "gpio54", "gpio68";
-			function = "gpio";
-			drive-strength = <2>;
-			bias-pull-up;
-		};
-	};
-
-	led_pins: led_pins {
-		mux {
-			pins = "gpio22", "gpio23", "gpio24";
-			function = "gpio";
-			drive-strength = <2>;
-			bias-pull-down;
-		};
-	};
-
-	rgmii2_pins: rgmii2_pins {
-		mux {
-			pins = "gpio27", "gpio28", "gpio29", "gpio30", "gpio31",
-				"gpio51", "gpio52", "gpio59", "gpio60", "gpio61", "gpio62";
-			function = "rgmii2";
-			drive-strength = <8>;
-			bias-disable;
-		};
-
-		tx {
-			pins = "gpio27", "gpio28", "gpio29", "gpio30", "gpio31", "gpio32";
-			input-disable;
-		};
-	};
-
-	spi_pins: spi_pins {
-		cs {
-			pins = "gpio20";
-			drive-strength = <12>;
-		};
-	};
-};
-
-&gsbi5 {
-	qcom,mode = <GSBI_PROT_SPI>;
-	status = "okay";
-
-	spi4: spi@1a280000 {
-		status = "okay";
-
-		pinctrl-0 = <&spi_pins>;
-		pinctrl-names = "default";
-
-		cs-gpios = <&qcom_pinmux 20 GPIO_ACTIVE_HIGH>;
-
-		m25p80@0 {
-			compatible = "everspin,mr25h256";
-			#address-cells = <1>;
-			#size-cells = <1>;
-			spi-max-frequency = <40000000>;
-			reg = <0>;
-		};
-	};
-};
-
-&nand_controller {
-	status = "okay";
-
-	pinctrl-0 = <&nand_pins>;
-	pinctrl-names = "default";
-
-	nand@0 {
-		reg = <0>;
-		compatible = "qcom,nandcs";
-
-		nand-ecc-strength = <4>;
-		nand-bus-width = <8>;
-		nand-ecc-step-size = <512>;
-
-		nand-is-boot-medium;
-		qcom,boot_pages_size = <0x1180000>;
-
-		partitions {
-			compatible = "fixed-partitions";
-			#address-cells = <1>;
-			#size-cells = <1>;
-
-			partition@0 {
-				label = "0:SBL1";
-				reg = <0x0000000 0x0040000>;
-				read-only;
-			};
-			partition@40000 {
-				label = "0:MIBIB";
-				reg = <0x0040000 0x0140000>;
-				read-only;
-			};
-			partition@180000 {
-				label = "0:SBL2";
-				reg = <0x0180000 0x0140000>;
-				read-only;
-			};
-			partition@2c0000 {
-				label = "0:SBL3";
-				reg = <0x02c0000 0x0280000>;
-				read-only;
-			};
-			partition@540000 {
-				label = "0:DDRCONFIG";
-				reg = <0x0540000 0x0120000>;
-				read-only;
-			};
-			partition@660000 {
-				label = "0:SSD";
-				reg = <0x0660000 0x0120000>;
-				read-only;
-			};
-			partition@780000 {
-				label = "0:TZ";
-				reg = <0x0780000 0x0280000>;
-				read-only;
-			};
-			partition@a00000 {
-				label = "0:RPM";
-				reg = <0x0a00000 0x0280000>;
-				read-only;
-			};
-			partition@c80000 {
-				label = "0:APPSBL";
-				reg = <0x0c80000 0x0500000>;
-				read-only;
-			};
-			partition@1180000 {
-				label = "0:APPSBLENV";
-				reg = <0x1180000 0x0080000>;
-			};
-			ART: partition@1200000 {
-				label = "0:ART";
-				reg = <0x1200000 0x0140000>;
-				read-only;
-			};
-			partition@1340000 {
-				label = "0:BOOTCONFIG";
-				reg = <0x1340000 0x0060000>;
-				read-only;
-			};
-			partition@13a0000 {
-				label = "0:SBL2_1";
-				reg = <0x13a0000 0x0140000>;
-				read-only;
-			};
-			partition@14e0000 {
-				label = "0:SBL3_1";
-				reg = <0x14e0000 0x0280000>;
-				read-only;
-			};
-			partition@1760000 {
-				label = "0:DDRCONFIG_1";
-				reg = <0x1760000 0x0120000>;
-				read-only;
-			};
-			partition@1880000 {
-				label = "0:SSD_1";
-				reg = <0x1880000 0x0120000>;
-				read-only;
-			};
-			partition@19a0000 {
-				label = "0:TZ_1";
-				reg = <0x19a0000 0x0280000>;
-				read-only;
-			};
-			partition@1c20000 {
-				label = "0:RPM_1";
-				reg = <0x1c20000 0x0280000>;
-				read-only;
-			};
-			partition@1ea0000 {
-				label = "0:BOOTCONFIG1";
-				reg = <0x1ea0000 0x0060000>;
-				read-only;
-			};
-			partition@1f00000 {
-				label = "0:APPSBL_1";
-				reg = <0x1f00000 0x0500000>;
-				read-only;
-			};
-			ubi: partition@2400000 {
-				label = "ubi";
-			};
-		};
-	};
-};
-
-&mdio0 {
-	status = "okay";
-
-	pinctrl-0 = <&mdio0_pins>;
-	pinctrl-names = "default";
-
-	phy0: ethernet-phy@0 {
-		reg = <0x0>;
-		qca,ar8327-initvals = <
-			0x00004 0x7600000   /* PAD0_MODE */
-			0x00008 0x1000000   /* PAD5_MODE */
-			0x0000c 0x80        /* PAD6_MODE */
-			0x000e4 0xaa545     /* MAC_POWER_SEL */
-			0x000e0 0xc74164de  /* SGMII_CTRL */
-			0x0007c 0x4e        /* PORT0_STATUS */
-			0x00094 0x4e        /* PORT6_STATUS */
-			0x00050 0xcf02cf02  /* LED_CTRL_0 */
-			0x00054 0xc832c832  /* LED_CTRL_1 */
-			>;
-	};
-	phy4: ethernet-phy@4 {
-		reg = <4>;
-	};
-};
-
-&gmac0 {
-	status = "okay";
-	compatible = "qcom,nss-gmac";
-	reg = <0x37000000 0x200000>;
-	interrupts = <GIC_SPI 220 IRQ_TYPE_LEVEL_HIGH>;
-	phy-mode = "rgmii";
-	qcom,id = <0>;
-	qcom,pcs-chanid = <0>;
-	qcom,phy-mdio-addr = <0>;
-	qcom,poll-required = <0>;
-	qcom,rgmii-delay = <1>;
-	qcom,phy_mii_type = <0>;
-	qcom,emulation = <0>;
-	qcom,forced-speed = <1000>;
-	qcom,forced-duplex = <1>;
-	qcom,socver = <0>;
-	qcom,irq = <255>;
-	mdiobus = <&mdio0>;
-
-	mtd-mac-address = <&ART 0x0>;
-
-	pinctrl-0 = <&rgmii2_pins>;
-	pinctrl-names = "default";
-
-	fixed-link {
-		speed = <1000>;
-		full-duplex;
-	};
-};
-
-&gmac1 {
-	status = "okay";
-	compatible = "qcom,nss-gmac";
-	reg = <0x37200000 0x200000>;
-	interrupts = <GIC_SPI 223 IRQ_TYPE_LEVEL_HIGH>;
-	phy-mode = "sgmii";
-	qcom,id = <1>;
-	qcom,pcs-chanid = <1>;
-	qcom,phy-mdio-addr = <4>;
-	qcom,poll-required = <0>;	/* no polling */
-	qcom,rgmii-delay = <0>;
-	qcom,phy_mii_type = <1>;
-	qcom,emulation = <0>;
-	qcom,forced-speed = <1000>;
-	qcom,forced-duplex = <1>;
-	qcom,socver = <0>;
-	qcom,irq = <258>;
-	mdiobus = <&mdio0>;
-
-	mtd-mac-address = <&ART 0x6>;
-
-	fixed-link {
-		speed = <1000>;
-		full-duplex;
-	};
-};
-
-&adm_dma {
-	status = "okay";
-};
-
-&usb3_0 {
-	status = "okay";
-	clocks = <&gcc USB30_1_MASTER_CLK>;
-};
-
-&usb3_1 {
-	status = "okay";
-	clocks = <&gcc USB30_0_MASTER_CLK>;
-};
-
-&pcie0 {
-	status = "okay";
-	reset-gpio = <&qcom_pinmux 3 GPIO_ACTIVE_HIGH>;
-	pinctrl-0 = <&pcie0_pins>;
-	pinctrl-names = "default";
-};
-
-&pcie1 {
-	status = "okay";
-	reset-gpio = <&qcom_pinmux 48 GPIO_ACTIVE_HIGH>;
-	pinctrl-0 = <&pcie1_pins>;
-	pinctrl-names = "default";
-	max-link-speed = <1>;
-};
diff --git a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev6.dts b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev6.dts
--- a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev6.dts	2021-08-13 03:43:31.636528074 +0000
+++ b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev6.dts	1970-01-01 00:00:00.000000000 +0000
@@ -1,13 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-or-later
-
-#include "qcom-ipq8065-rt4230w.dtsi"
-#include <dt-bindings/input/input.h>
-
-/ {
-	model = "Askey RT4230W REV6";
-	compatible = "askey,rt4230w-rev6", "qcom,ipq8065", "qcom,ipq8064";
-};
-
-&ubi {
-	reg = <0x2400000 0x1a000000>;
-};
diff --git a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev9.3.dts b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev9.3.dts
--- a/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev9.3.dts	2021-08-13 03:43:31.636528074 +0000
+++ b/target/linux/ipq806x/files/arch/arm/boot/dts/qcom-ipq8065-rt4230w-rev9.3.dts	1970-01-01 00:00:00.000000000 +0000
@@ -1,13 +0,0 @@
-// SPDX-License-Identifier: GPL-2.0-or-later
-
-#include "qcom-ipq8065-rt4230w.dtsi"
-#include <dt-bindings/input/input.h>
-
-/ {
-	model = "Askey RT4230W REV9.3";
-	compatible = "askey,rt4230w-rev9.3", "qcom,ipq8065", "qcom,ipq8064";
-};
-
-&ubi {
-	reg = <0x2400000 0xdc00000>;
-};
diff --git a/target/linux/ipq806x/image/Makefile b/target/linux/ipq806x/image/Makefile
--- a/target/linux/ipq806x/image/Makefile	2021-08-13 03:43:31.636528074 +0000
+++ b/target/linux/ipq806x/image/Makefile	2021-08-13 04:10:56.592604048 +0000
@@ -101,29 +101,6 @@
 	IMAGE/sysupgrade.bin/squashfs := append-rootfs | pad-to $$$$(BLOCKSIZE) | sysupgrade-tar rootfs=$$$$@ | append-metadata
 endef
 
-define Device/askey_rt4230w
-	$(call Device/LegacyImage)
-	DEVICE_VENDOR := Askey
-	DEVICE_MODEL := RT4230W
-	SOC := qcom-ipq8065
-	BLOCKSIZE := 128k
-	PAGESIZE := 2048
-	KERNEL_IN_UBI := 1
-	DEVICE_PACKAGES := ath10k-firmware-qca9984-ct
-endef
-
-define Device/askey_rt4230w-rev6
-	$(call Device/askey_rt4230w)
-	DEVICE_VARIANT := REV6
-endef
-TARGET_DEVICES += askey_rt4230w-rev6
-
-define Device/askey_rt4230w-rev9.3
-	$(call Device/askey_rt4230w)
-	DEVICE_VARIANT := REV9.3
-endef
-TARGET_DEVICES += askey_rt4230w-rev9.3
-
 define Device/asrock_g10
 	$(call Device/FitImage)
 	$(call Device/UbiFit)
@@ -307,6 +284,23 @@
 endef
 TARGET_DEVICES += netgear_r7500v2
 
+define Device/askey_rt4230w
+	$(call Device/DniImage)
+ 	DEVICE_VENDOR := Askey
+ 	DEVICE_MODEL := RT4230W
+	DEVICE_DTS := qcom-ipq8065-rt4230w
+ 	SOC := qcom-ipq8065
+	KERNEL_SIZE := 4096k
+	NETGEAR_BOARD_ID := RT4230W
+	NETGEAR_HW_ID := 29764958+0+512+1024+4x4+4x4+cascade
+ 	BLOCKSIZE := 128k
+ 	PAGESIZE := 2048
+	BOARD_NAME := rt4230w
+	SUPPORTED_DEVICES += rt4230w
+	DEVICE_PACKAGES := ath10k-firmware-qca9984-ct kmod-qca-nss-gmac
+ endef
+TARGET_DEVICES += askey_rt4230w
+
 define Device/netgear_r7800
 	$(call Device/DniImage)
 	DEVICE_VENDOR := NETGEAR
diff --git a/target/linux/ipq806x/patches-5.4/0069-arm-boot-add-dts-files.patch b/target/linux/ipq806x/patches-5.4/0069-arm-boot-add-dts-files.patch
--- a/target/linux/ipq806x/patches-5.4/0069-arm-boot-add-dts-files.patch	2021-08-13 03:43:31.640528063 +0000
+++ b/target/linux/ipq806x/patches-5.4/0069-arm-boot-add-dts-files.patch	2021-08-13 04:09:18.424530361 +0000
@@ -10,7 +10,7 @@
 
 --- a/arch/arm/boot/dts/Makefile
 +++ b/arch/arm/boot/dts/Makefile
-@@ -842,7 +842,29 @@ dtb-$(CONFIG_ARCH_QCOM) += \
+@@ -842,7 +842,28 @@ dtb-$(CONFIG_ARCH_QCOM) += \
  	qcom-ipq4019-ap.dk04.1-c3.dtb \
  	qcom-ipq4019-ap.dk07.1-c1.dtb \
  	qcom-ipq4019-ap.dk07.1-c2.dtb \
@@ -34,8 +34,7 @@
 +	qcom-ipq8065-nbg6817.dtb \
 +	qcom-ipq8065-r7800.dtb \
 +	qcom-ipq8065-xr500.dtb \
-+	qcom-ipq8065-rt4230w-rev6.dtb \
-+	qcom-ipq8065-rt4230w-rev9.3.dtb \
++	qcom-ipq8065-rt4230w.dtb \
 +	qcom-ipq8068-ecw5410.dtb \
  	qcom-msm8660-surf.dtb \
  	qcom-msm8960-cdp.dtb \

